### PR TITLE
fix kubectl set image --record option 

### PR DESCRIPTION
**Pull request by WanLinghao**
In kubectl set image command, if once set 'record' option to betrue, it will always record user's command even when user set 'record' option to be false.
For example:
1.kubectl set image deployment/test-deployment test=nginx:1.9.1 --record=true
2.kubectl set image deployment/test-deployment test=nginx:1.9.2 --record=false
1 and 2 will both record user's command.

this patch fix this.after this patch:
1.kubectl set image deployment/test-deployment test=nginx:1.9.1 --record=true
2.kubectl set image deployment/test-deployment test=nginx:1.9.2 --record=false
command 1 will record but command will not record.

modified:   pkg/kubectl/cmd/set/set_image.go
modified:   pkg/kubectl/cmd/util/helpers.go

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```
NONE
```

### CODE REVIEW

**Comment by WanLinghao** [comment]
@juanvallejo  @mengqiy PTAL

File: pkg/kubectl/cmd/util/helpers.go [hunk 1]
```diff
@@ -542,6 +542,21 @@ func RecordChangeCause(obj runtime.Object, changeCause string) error {
 	return nil
 }
 
+// RemoveAnnotations remove change-cause annotation in input runtime object
+func RemoveAnnotations(obj runtime.Object) error {
```
Comment (juanvallejo) [inline pkg/kubectl/cmd/util/helpers.go]: s/RemoveAnnotations/RemoveChangeCauseAnnotations/

File: pkg/kubectl/cmd/util/helpers.go [hunk 2]
```diff
@@ -573,6 +588,37 @@ func ChangeResourcePatch(info *resource.Info, changeCause string) ([]byte, types
 	}
 }
 
+// RemoveAnnotationsPatch craetes a patch between the origin input resource info
+// and the input resource info without change-cause annotation
+func RemoveAnnotationsPatch(info *resource.Info) ([]byte, types.PatchType, error) {
```
Comment (juanvallejo) [inline pkg/kubectl/cmd/util/helpers.go]: To keep in-line with the name of related helpers, maybe rename to RemoveChangeResourcePatch

File: pkg/kubectl/cmd/set/set_image.go [hunk 3]
```diff
@@ -251,24 +251,40 @@ func (o *ImageOptions) Run() error {
...
 					fmt.Fprintf(o.Err, "WARNING: changes to %s/%s can't be recorded: %v\n", info.Mapping.Resource, info.Name, err)
 				}
 			}
-		}
+			info.Refresh(obj, true)
+		} else {
+			//patch but NOT record the change
+			if patch, patchType, err := cmdutil.RemoveAnnotationsPatch(info); err == nil {
+				if obj, err = resource.NewHelper(info.Client, info.Mapping).Patch(info.Namespace, info.Name, patchType, patch); err != nil {
+					fmt.Fprintf(o.Err, "WARNING: changes to %s/%s can't be recorded: %v\n", info.Mapping.Resource, info.Name, err)
```
Comment (juanvallejo) [inline pkg/kubectl/cmd/set/set_image.go]: error should say something similar to `unable to remove change-cause annotation`

File: pkg/kubectl/cmd/set/set_image.go [hunk 4]
```diff
@@ -251,24 +251,40 @@ func (o *ImageOptions) Run() error {
...
-		if err != nil {
-			allErrs = append(allErrs, fmt.Errorf("failed to patch image update to pod template: %v\n", err))
-			continue
-		}
-		info.Refresh(obj, true)
+		var obj runtime.Object
+		var err error
+
+		if cmdutil.ShouldRecord(o.Cmd, info) {
+			//patch and record the change(for rollout history)
```
Comment (juanvallejo) [inline pkg/kubectl/cmd/set/set_image.go]: nit: include a space between `//` and the rest of the comment

**Comment by juanvallejo** [comment]
cc @fabianofranz

**Comment by WanLinghao** [comment]
@juanvallejo  thanks for your reply. done

**Comment by WanLinghao** [comment]
@liggitt  PTAL

**Comment by WanLinghao** [comment]
@juanvallejo  please tell me who can help me push this patch, thanks a lot

**Comment by juanvallejo** [comment]
@mfojtik friendly ping

**Comment by WanLinghao** [comment]
@sttts @adohe PTAL

**Pull request closed without merging**